src/module.c

```diff
@@ -10630,11 +10754,25 @@ int moduleLoad(const char *path, void **module_argv, int module_argc) {
     RedisModuleCtx ctx;
     moduleCreateContext(&ctx, NULL, REDISMODULE_CTX_TEMP_CLIENT); /* We pass NULL since we don't have a module yet. */
     selectDb(ctx.client, 0);
+
+    /* If this is a loadex command we want to populate server.module_configs_queue with sds NAME VALUE pairs.
+     * We also want to increment argv to just after ARGS. */
+    if (is_loadex) {
+        /* This call will also increment the module_argv pointer to just after ARGS. */
+        if (parseLoadexArguments((RedisModuleString ***) &module_argv, &module_argc)) {
```

Comment:
We should parse these parameters "before" starting to do any loading. Failing here for syntax errors just makes life more complex.